### PR TITLE
LLT-3737: Add more upnp integrational test

### PR DESCRIPTION
**Pull request by arvydasv**
Re-enable/cleanup direct connection tests on natlab.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean
- [x] README.md is updated
- [x] changelog.md is updated

### CODE REVIEW

File: nat-lab/tests/test_direct_connection.py [hunk 1]
```diff
@@ -32,7 +47,7 @@
         STUN_PROVIDER,
         ConnectionTag.DOCKER_SYMMETRIC_CLIENT_1,
         ConnectionTag.DOCKER_FULLCONE_CLIENT_1,
-        DOCKER_SYMMETRIC_GW_1_IP,
```
Comment (Jauler) [inline nat-lab/tests/test_direct_connection.py:35]: Is this change really intended? If yes, why should we remove symmetric NAT <-> fullcone NAT test?

File: nat-lab/tests/test_direct_connection.py [hunk 2]
```diff
@@ -76,90 +91,221 @@
...
+        self.conn = conn
+        self.conn_track = conn_track
+        self.conn_gw = conn_gw
+
+
+@asynccontextmanager
+async def new_connections_with_mesh_clients(
+    exit_stack: AsyncExitStack,
+    client1_type: ConnectionTag,
+    ep1: List[str],
```
Comment (Jauler) [inline nat-lab/tests/test_direct_connection.py:123]: nitpick: Would it be possible to rename ep[123] to something that specifies that this is not the first endpoint, but a list of endpoint providers for first node? During the reading of the code I got confused about this part, and this may be nice readability improvement.

But consider this note as low prio, and if you think, that it makes the code cumbersome, feel free to just mark the note as "resolved" yourself.

File: nat-lab/tests/test_direct_connection.py (hunk 1 above)
Comment (Jauler) [reply nat-lab/tests/test_direct_connection.py:35]: Oh wait I got confused about arguments, this is reflexive IP, not the NAT type. Ignore this comment.

Comment (arvydasv) [reply nat-lab/tests/test_direct_connection.py:35]: In the test the first client ( Symetric ) will try to ping client two ( Fullcone ) while the second client's reflexive ip path is disable.

**Review by tomaszklak — COMMENTED** [review]
+1.0

File: nat-lab/tests/test_upnp_connection.py [hunk 3]
```diff
@@ -68,27 +68,27 @@ async def test_upnp_route_corrupted() -> None:
...
         await beta_connection_gw.create_process(["conntrack", "-F"]).execute()
-        await alpha_connection_gw.create_process(["upnpd", "eth0", "eth1"]).execute()
-        await beta_connection_gw.create_process(["upnpd", "eth0", "eth1"]).execute()
 
         with pytest.raises(asyncio.TimeoutError):
             async with Ping(alpha_connection, beta.ip_addresses[0]).run() as ping:
                 await testing.wait_long(ping.wait_for_next_ping())
 
+        await alpha_connection_gw.create_process(["upnpd", "eth0", "eth1"]).execute()
+        await beta_connection_gw.create_process(["upnpd", "eth0", "eth1"]).execute()
```
[the remark this answers is not in the document]
Comment (arvydasv) [reply nat-lab/tests/test_upnp_connection.py:82]: I changed the test itself to "removed" from "corrunted"

[the remark this answers is not in the document]
Comment (tomaszklak) [reply nat-lab/tests/test_upnp_connection.py:82]: I think a comment explaining **why** this has to be after the ping would be a good idea.

[the remark this answers is not in the document]
Comment (tomaszklak) [reply nat-lab/tests/test_upnp_connection.py:82]: The comments you've added explain **what** was done, not **why** it was done.

[the remark this answers is not in the document]
Comment (arvydasv) [reply nat-lab/tests/test_upnp_connection.py:82]: Done.

File: nat-lab/tests/test_direct_connection.py [hunk 4]
```diff
@@ -76,90 +92,223 @@
         ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_1,
         DOCKER_OPEN_INTERNET_CLIENT_1_IP,
     ),
+    (
+        UPNP_PROVIDER,
+        ConnectionTag.DOCKER_UPNP_CLIENT_1,
+        ConnectionTag.DOCKER_UPNP_CLIENT_2,
+        DOCKER_UPNP_CLIENT_2_IP,
+    ),
```
Comment (Jauler) [inline nat-lab/tests/test_direct_connection.py:100]: We should also add UPNP provider where it is available only on one side, e.g. UPnP<->CONE

Comment (Jauler) [reply nat-lab/tests/test_direct_connection.py:100]: other than that looks good -> +1.0

Comment (arvydasv) [reply nat-lab/tests/test_direct_connection.py:100]: That makes it +2 with Tomasz. Could you approve the PR? :)

File: nat-lab/tests/test_direct_connection.py [hunk 5]
```diff
@@ -76,90 +92,223 @@
...
+                ping_limits=ConnectionLimits(0, 5),
+            ),
+        )
+    )
+
+    alpha_client = await exit_stack.enter_async_context(
+        telio.Client(
+            alpha_conn,
+            alpha,
+            telio.AdapterType.BoringTun,
```
Comment (lcruz99) [inline nat-lab/tests/test_direct_connection.py:178]: Is it required to use BoringTun? On another tests the adapter is ommited to telio.Client constructor, leaving it as AdapterType.Default

**Review by Jauler — DISMISSED** [review]
+1.0

File: nat-lab/tests/utils/connection_tracker.py [hunk 6]
```diff
@@ -162,6 +162,9 @@ async def wait_for_event(self, key: str) -> None:
             self._events_wait_list[key].append(event)
         await event.wait()
 
+    async def clear(self) -> None:
```
[the remark this answers is not in the document]
Comment (gytsto) [reply nat-lab/tests/utils/connection_tracker.py:165]: just do 
```
await alpha.create_process(["conntrack", "-F"]).execute()
```

since it is only used in one place

[the remark this answers is not in the document]
Comment (arvydasv) [reply nat-lab/tests/utils/connection_tracker.py]: Done

**Review by gytsto — APPROVED** [review]
gj +1

**Review by gytsto — APPROVED** [review]
